ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/highLevelHumanoidControl/highLevelStates/walkingController/states/WalkingSingleSupportState.java

```diff
@@ -311,28 +332,67 @@ public void onEntry()
          balanceManager.addFootstepToPlan(footsteps[i], footstepTimings[i]);
       }
 
-      balanceManager.setICPPlanSupportSide(supportSide);
-      balanceManager.initializeICPPlanForSingleSupport();
 
-      /** This has to be called after calling initialize ICP Plan, as that resets the step constraints **/
-      walkingMessageHandler.pollStepConstraints(stepConstraints);
-      balanceManager.setCurrentStepConstraints(stepConstraints);
 
+      //
       updateHeightManager();
 
-      feetManager.requestSwing(swingSide,
-                               nextFootstep,
-                               swingTime,
-                               balanceManager.getFinalDesiredCoMVelocity(),
-                               balanceManager.getFinalDesiredCoMAcceleration());
+      //
+      if (firstTick)
+      {
+         feetManager.requestSwing(swingSide,
+                                  nextFootstep,
+                                  swingTime,
+                                  balanceManager.getFinalDesiredCoMVelocity(),
+                                  balanceManager.getFinalDesiredCoMAcceleration());
 
+      }
+
+      else
+         feetManager.adjustSwingTrajectory(swingSide, nextFootstep, swingTime);
+
+      //
       if (feetManager.adjustHeightIfNeeded(nextFootstep))
       {
          walkingMessageHandler.updateVisualizationAfterFootstepAdjustement(nextFootstep);
          feetManager.adjustSwingTrajectory(swingSide, nextFootstep, swingTime);
       }
 
+      //
+      controllerToolbox.updateBipedSupportPolygons();
+
       balanceManager.setSwingFootTrajectory(swingSide, feetManager.getSwingTrajectory(swingSide));
+      balanceManager.adjustFootstepInCoPPlan(nextFootstep);
+      balanceManager.computeICPPlan();
+   }
+
+   private boolean haveWeEntered = false;
```

Comment:
I am confused by this boolean. Isn't whether or not you are executing the method inside `onEntry()` enough information to set `firstTick `in the method? Why are `haveWeEntered `and `firstTick `both in `handleNewFootstep()`?

Reply:
There's probably a better way to do this whole thing. They don't exactly convey the same information, since first tick could be false if it is the second+ tick or if it has not entered the state yet, and that distinction is important. It is possible for the method that relies on this boolean (`handleNewFootstep()`) to be called without entering the state (`WalkingSingleSupportState`) because I have this method tied to a listener in another class (`WalkingMessageHandler`), and the class that this method belongs to is instantiated twice, once for each leg. So it is possible for this to be called by the listener even if the class (and therefore single support of the proper leg) has not been called.

The thing is, this is probably redundant because I later ended up adding a check to make sure that we only proceed with the `handleNewFootstep()` method if we are on the proper side of the robot. So I can probably delete the logic that relies on `haveWeEntered`